pkg/ibmvpc/nacl_analysis.go

```diff
@@ -92,26 +95,9 @@ func (na *IBMNACLAnalyzer) GetNACLRule(index int) (ruleStr string, ruleRes *comm
 	return ruleStr, ruleRes, isIngress, nil
 }
 
+// GetNACLRules returns ingress and egress rule objects
 func (na *IBMNACLAnalyzer) GetNACLRules() (ingressRules, egressRules []*commonvpc.NACLRule, err error) {
-	ingressRules = []*commonvpc.NACLRule{}
-	egressRules = []*commonvpc.NACLRule{}
-	for index := range na.naclResource.Rules {
-		rule := na.naclResource.Rules[index]
-		_, ruleObj, isIngress, err := na.GetNACLRule(index)
-		if err != nil {
-			return nil, nil, err
-		}
-		if rule == nil {
-			continue
-		}
-		na.referencedIPblocks = append(na.referencedIPblocks, ruleObj.Src.Split()...)
-		na.referencedIPblocks = append(na.referencedIPblocks, ruleObj.Dst.Split()...)
-		ruleObj.Index = index
-		if isIngress {
-			ingressRules = append(ingressRules, ruleObj)
-		} else {
-			egressRules = append(egressRules, ruleObj)
-		}
-	}
-	return ingressRules, egressRules, nil
+	ingressRules, egressRules, referencedIPblocks, err := commonvpc.GetNACLRules(na.GetNumberOfRules(), na)
+	na.referencedIPblocks = referencedIPblocks
```

Comment:
SpecificNACLAnalyzer should implement SetReferencedIPblocks()
SetReferencedIPblocks() should be called inside commonvpc.GetNACLRules()